tests/func/experiments/test_experiments.py

```diff
@@ -624,3 +624,47 @@ def test_fix_exp_head(tmp_dir, scm, tail):
     scm.set_ref(EXEC_BASELINE, "refs/heads/master")
     assert EXEC_BASELINE + tail == fix_exp_head(scm, head)
     assert "foo" + tail == fix_exp_head(scm, "foo" + tail)
+
+
+@pytest.mark.parametrize("workspace", [True, False])
+def test_modified_data_dep(tmp_dir, scm, dvc, workspace):
+    tmp_dir.dvc_gen("data", "data")
+    tmp_dir.gen("copy.py", COPY_SCRIPT)
+    tmp_dir.gen("params.yaml", "foo: 1")
+    exp_stage = dvc.run(
+        cmd="python copy.py params.yaml metrics.yaml",
+        metrics_no_cache=["metrics.yaml"],
+        params=["foo"],
+        name="copy-file",
+        deps=["copy.py", "data"],
+    )
+    scm.add(
+        [
+            "dvc.yaml",
+            "dvc.lock",
+            "copy.py",
+            "params.yaml",
+            "metrics.yaml",
+            "data.dvc",
+            ".gitignore",
+        ]
+    )
+    scm.commit("init")
+
+    tmp_dir.gen("params.yaml", "foo: 2")
```

Comment:
Maybe include a test where params don't change (only data)?

Reply:
Also maybe test multiple modified data deps?

Reply:
Updated so we test with and without params changes (so we ensure that a new exp branch can be triggered solely on modified data deps).

Testing w/multiple data deps should not be necessary here, we are re-using the existing `repo.commit`/`stage.commit` functionality, handling changes in multiple deps is already tested at the `commit` level elsewhere.